### PR TITLE
feat: custom starter message configuration

### PR DESCRIPTION
**Pull request by abeatrix**
RE: https://sourcegraph.slack.com/archives/C05AGQYD528/p1693983800785199

This PR adds a new configuration `cody.customer.starter`  to allow setting a custom starter message that will prepend chat requests to Cody:
- Configuration now has a `cody.customer.starter` string property.
- A new setting cody.custom.starter was added in vscode/package.json to allow configuring a custom message.
- getConfiguration() was updated to read the new customStarter config value.
- ConfigurationWithAccessToken gained a customStarter property.
- main.ts imports PromptMixin to set the custom starter message on extension start.

## Test plan

1. set `cody.customer.starter` in your user config, e.g. "call me jeff"
2. reload extension
3. submit a chat request
4. check the output channel to see your setting was added to the chat request

Screenshot 2023-09-06 at 9 03 38 PM

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -811,6 +811,14 @@
           "default": false,
           "markdownDescription": "Experimental feature for internal use."
         },
+        "cody.custom.starter": {
```
Comment (valerybugakov) [inline vscode/package.json:814]: Many settings are grouped under feature categories (e.g., `cody.autocomplete.*` or `cody.inlineChat.*`). Would it make sense to apply this pattern here too?

**Review by valerybugakov — APPROVED** [review]

File: vscode/package.json (hunk 1 above)
Comment (toolmantim) [reply vscode/package.json:814]: What about `cody.chat.preInstruction` with the description "An instruction for Cody to be included at the start of all chat messages sent"

That is assuming it's chat messages only?

**Review by toolmantim — APPROVED** [review]
Great! I left a suggestion for a new config name.

File: vscode/package.json (hunk 1 above)
Comment (abeatrix) [reply vscode/package.json:814]: Yea autocompletes uses a different prompt so this is only for chat

**Pull request merged**